packages/core/src/base-task.ts

```diff
@@ -2,28 +2,53 @@ import * as debug from 'debug';
 
 import { TaskContext, TaskIdentifier, TaskMetaData } from './types/tasks';
 
+import { TaskConfig } from './types/configuration';
+
 export default abstract class BaseTask {
   context: TaskContext;
   meta!: TaskMetaData | TaskIdentifier;
   debug: debug.Debugger;
-  private _options!: boolean | object;
+
+  private _config!: unknown;
+  private _enabled!: string;
```

Comment:
Can we just use private fields? Seems like TS 3.8 added support for them, and they would enforce this pretty nicely...

```suggestion
  private #config: unknown;
  private #enabled: string;
```

Some prosy bits: https://devblogs.microsoft.com/typescript/announcing-typescript-3-8-beta/#ecmascript-private-fields